webapps/api/backend/decorators.py

```diff
@@ -257,21 +259,32 @@ def wrap(f):
         def new_function(*args, **kwargs):
             """
             Wrapped function
+            This function will process the api request an apply:
+             - Paging (only return a subset of all results)
+             - Sorting (sort on properties)
+             - Filtering (filter on properties)
+            Request arguments for paging:
+            - page: The page number for which the items should be displayed (string/int)
+            - page_size: The size of the pages (string/int)
+            Request arguments for sorting:
+            - sort: Comma separated list of the properties to sort on. Prefix with '-' to use descending order (eg name,-description) (string)
+            Request arguments for filtering: identical to DataList query params
+            - query: The query to perform. See DataList execute_query method for more info
             """
             request = _find_request(args)
             timings = {}
 
             # 1. Pre-loading request data
             start = time.time()
             sort = request.QUERY_PARAMS.get('sort')
+            query = request.QUERY_PARAMS.get('query')
             if sort is None and default_sort is not None:
                 sort = default_sort
             sort = None if sort is None else [s for s in reversed(sort.split(','))]
             page = request.QUERY_PARAMS.get('page')
-            page = int(page) if page is not None and page.isdigit() else None
+            page = int(page) if page is not None and (isinstance(page, int) or page.isdigit()) else None
             page_size = request.QUERY_PARAMS.get('page_size')
-            page_size = int(page_size) if page_size is not None and page_size.isdigit() else None
-            page_size = page_size if page_size in [10, 25, 50, 100] else 10
```

Comment:
Why did you remove these default sizes?
Our GUI only supports these values, so shouldn't we keep this?

Reply:
Our GUI could support other sizes too. Our dropdown is only limited to those.
No need to keep the defaults as it is the callee's problem when their pagination is not returning their number of items

Also it is a bit nasty to not reject but instead go by the default of 10